src/nvim/window.c

```diff
@@ -4647,9 +4659,30 @@ win_free (
   /* Remove the window from the b_wininfo lists, it may happen that the
    * freed memory is re-used for another window. */
   FOR_ALL_BUFFERS(buf) {
-    for (wip = buf->b_wininfo; wip != NULL; wip = wip->wi_next)
-      if (wip->wi_win == wp)
+    for (wip = buf->b_wininfo; wip != NULL; wip = wip->wi_next) {
+      if (wip->wi_win == wp) {
+        wininfo_T *wip2;
+
+        // If there already is an entry with "wi_win" set to NULL it
+        // must be removed, it would never be used.
+        for (wip2 = buf->b_wininfo; wip2 != NULL; wip2 = wip2->wi_next) {
```

Comment:
Where in the loop does it depend on the current `wip`?

Reply:
The code in the loop is independent of `wip`.

Reply:
why is it running in this loop?

Reply:
I guess the intent to have that in the loop is to have it as part of the `if` clause, whose condition is a function of `wip`, but I'm not sure.